controller/control.go

```diff
@@ -135,8 +137,18 @@ func (c *Controller) UpdateVolStatus() {
 		c.ReadOnly = true
 	}
 
-	logrus.Infof("Previously Volume RO: %v, Currently: %v,  Total Replicas: %v,  RW replicas: %v, Total backends: %v",
-		prev, c.ReadOnly, len(c.replicas), rwReplicaCount, len(c.backend.backends))
+	if rwReplicaCount == c.ReplicationFactor {
```

Comment:
lets use this function to just UpdateVolStatus.
Make a new function for this code, and call it in the caller with probably any conditions required.
You can have new conditions in new function also.

Reply:
done